validate.go

```diff
@@ -152,6 +170,12 @@ func mungeValidateErr(err error, sch Schema) error {
 				break
 			}
 
+			if schkind == cue.IntKind || schkind == cue.FloatKind || schkind == cue.NumberKind {
```

Comment:
i'd suggest doing a bitwise comparison here - i never quite trust that we get a value where it's _just_ one of these, so this covers the bases better

```suggestion
			if schkind & (cue.IntKind | cue.FloatKind | cue.NumberKind) != 0 {
```

Reply:
Diving deeper, I just realized that `NumberKind Kind = IntKind | FloatKind`, so I guess we could just do: `schkind & cue.NumberKind`, which is equivalent, and even simpler, right? 🤔 cc/ @undef1nd @K-Phoen